Plane/Copter/Rover/Tracker: 4.6.0-beta4 release

rmackay9 opened the pull request:
This is the 4.6.0-beta4 release for Copter, Rover, Tracker and Plane.

The full list of PRs can be seen in the "4.6.0-beta4" column of [this project](https://github.com/orgs/ArduPilot/projects/14/views/2).

There was just one PR that I was unable to backport due to a merge conflict but if we're keen to get this into 4.6 maybe @bugobliterator could produce a backport PR?

- https://github.com/ArduPilot/ardupilot/pull/28913

Any and all feedback on the release note wording or anything else is greatly appreciated!

Hwurzburg commented on AntennaTracker/ReleaseNotes.txt:
```diff
@@ -1,6 +1,35 @@
+
+1) Board specfic changes
+
+- BETAFPV F405 supports DPS310 baro
```
- Add BETAFPV F405 board variants

rmackay9 replied:
txs, added

rmackay9 commented on AntennaTracker/ReleaseNotes.txt:
```diff
@@ -1,6 +1,35 @@
+- AM32 ESC passthrough fixed
+- BMP581 baro initialisation fix
+- DDS/ROS2 driver waits indefinitely until companion computer running ROS2 starts
+- Lua script potential deadlock fix when receiving mavlink message
```
txs, fixed

rmackay9 commented on AntennaTracker/ReleaseNotes.txt:
```diff
@@ -1,6 +1,35 @@
+- DDS/ROS2 driver waits indefinitely until companion computer running ROS2 starts
+- Lua script potential deadlock fix when receiving mavlink message
+- Plane's TILT log message only logged on tilt-rotors
+- ROMFS race condition fix when opening files
```
added txs

peterbarker commented:
I've created a backport PR for the ibus naming fix  (https://github.com/ArduPilot/ardupilot/pull/29294) and removed the conflicting backport

peterbarker commented:
I've created a backport PR for the CaddX support and added it to the project, removing the old PR which was against master.

tridge reviewed (approved):
thanks Randy!

rmackay9 commented:
I've rebased and updated this PR after these backport were created and merged thanks to the kind help of @bugobliterator, @tpwrules and @peterbarker.

- https://github.com/ArduPilot/ardupilot/pull/29297
- https://github.com/ArduPilot/ardupilot/pull/29294
- https://github.com/ArduPilot/ardupilot/pull/29296

If I could just get one more approval, perhaps from @peterbarker then I'll merge and we can do the beta release

rmackay9 commented:
I've force pushed this one last time in order to include the CADDX driver fix https://github.com/ArduPilot/ardupilot/pull/29309

I plan to merge after this PR passes CI
